src/hv_aic_events.c

```diff
@@ -0,0 +1,166 @@
+/* SPDX-License-Identifier: MIT */
+
+#include "adt.h"
+#include "hv.h"
+#include "uartproxy.h"
+
+#define AIC_EVENT      0x2004
+#define AIC_EVENT_TYPE GENMASK(31, 16)
+#define AIC_EVENT_NUM  GENMASK(15, 0)
+
+#define AIC_EVENT_TYPE_HW   1
+#define AIC_EVENT_TYPE_IPI  4
+#define AIC_EVENT_IPI_OTHER 1
+#define AIC_EVENT_IPI_SELF  2
+
+#define AIC_MAX_HW_NUM 1024
+
+static u64 aic_base;
+
+static u8 trace_hw_num[AIC_MAX_HW_NUM / 8];
```

Comment:
Since AIC works in terms of 32bit registers for masks/unmasks, let's also have the masks use u32 units. Doesn't matter much here, but for mask/unmask tracing (which would use other arrays that don't exist yet) that means we can use boolean ops to check if any interesting IRQ was masked/unmasked in one go.